typed-racket-lib/typed-racket/utils/opaque-object.rkt

```diff
@@ -60,19 +62,19 @@
       (remove* fields actual-fields))
     (define remaining-methods
       (remove* methods actual-methods))
-    (define guard/c
-      (dynamic-object/c (append methods remaining-methods)
-                        (append method-ctcs
-                                (for/list ([m remaining-methods])
-                                  (restrict-typed->/c m)))
-                        (append fields remaining-fields)
-                        (append field-ctcs
-                                (for/list ([m remaining-fields])
-                                  (restrict-typed-field/c m)))))
+    (define restrict-guard/c
+      (dynamic-object/c remaining-methods
```

Comment:
If `remaining-methods` and `remaining-fields` are empty lists, then could we skip making & applying this `dynamic-object/c`?

Reply:
good idea, thanks.

Do you know how to tell if there is test coverage?